common/app/views/support/HtmlCleaner.scala

```diff
@@ -602,3 +603,24 @@ object MembershipEventCleaner extends HtmlCleaner {
     document
   }
 }
+
+object ChaptersLinksCleaner extends HtmlCleaner {
+  def slugify(text: String): String = {
+    Normalizer.normalize(text, Normalizer.Form.NFKD)
+      .toLowerCase
+      .replaceAll("[^0-9a-z ]", "")
+      .trim.replaceAll(" +", "-")
+  }
+
+  override def clean(document: Document): Document = {
+    if(ChapterHeadingsSwitch.isSwitchedOn) {
+      val autoaChapters = document.getElementsByClass("auto-chapter")
+
+      autoaChapters.foreach { ch =>
+        val h2 = ch.getElementsByTag("h2")
+        h2.attr("id", slugify(h2.text())).attr("class", "anchor-link-fix")
```

Comment:
@OliverJAsh it wouldn't even need to be all H2s, just H2s within sections with the class "auto-chapter". 

I did it this way though so this fix could be applied to other anchor link elements. e.g. the comment box at the bottom of the page.